File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    length = len(s)
    cnt = 0
    for i in range(length):
        if(s[i] == '('):
            cnt += 1
        else:
            cnt -= 1
            if(cnt < 0):
                answer = False
                break
    if(cnt != 0):
        answer = False
    return answer

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.Stack;

class Solution {
    

    public Boolean solution(String s) {
        Boolean answer = true;
        Stack<Character> st = new Stack<>();
        char[] c = s.toCharArray();
        for (char ch : c) {
            if (ch == '(') {
                st.add(ch);
            } else {
                if (st.size() == 0) {
                    answer = false;
                    break;
                }
                st.pop();
            }
        }
        if (st.size() != 0) {
            answer = false;
        }
        return answer;
    }
}